tools/SetupFlow/DevHome.SetupFlow/Models/ConfigureTargetTask.cs

```diff
@@ -499,4 +510,15 @@ public async Task CreateCorrectiveActionPanel(IExtensionAdaptiveCardSession2 ses
 
         return (_stringResource.GetLocalized(StringResourceKey.ConfigurationUnitSummaryFull, unit.Intent, unit.Type, packageId, unitDescription), packageName);
     }
+
+    private void LogCompletionTelemetry(TaskFinishedState taskFinishedState)

```

Comment:
Could directly send the EnvironmentsTelemetryStatus. And using this function at line 382 would be better.